tests/end-to-end/backup-restore-test/backupe2e/prometheus.go

```diff
@@ -0,0 +1,222 @@
+package backupe2e
+
+// Test http request to the prometheus api before backup and after restore.
+// An expected result at a specific point in time is considered a succeeded test.
+//
+// {
+//   "status":"success",
+//   "data":{
+//      "resultType":"vector",
+//      "result":[
+//         {
+//           "metric":{},
+//           "value":[
+//               1551421406.195,
+//               "1.661"
+//            ]
+//          }
+//      ]
+//    }
+// }
+//
+// {success {vector [{map[] [1.551424874014e+09 1.661]}]}}
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/google/uuid"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+const (
+	domain       = "http://monitoring-prometheus.kyma-system"
+	prometheusNS = "kyma-system"
+	api          = "/api/v1/query?"
+	metricsQuery = "max(sum(kube_pod_container_resource_requests_cpu_cores) by (instance))"
+	port         = "9090"
+	metricName   = "kube_pod_container_resource_requests_cpu_cores"
+)
+
+type queryResponse struct {
+	Status string       `json:"status"`
+	Data   responseData `json:"data"`
+}
+
+type responseData struct {
+	ResultType string       `json:"resultType"`
+	Result     []dataResult `json:"result"`
+}
+
+type dataResult struct {
+	Metric interface{}   `json:"metric,omitempty"`
+	Value  []interface{} `json:"value,omitempty"`
+}
+
+type prometheusTest struct {
+	metricName, uuid string
+	beforeBackup     queryResponse
+	afterRestore     queryResponse
+	expectedResult   string
+	finalResult      string
+	apiQuery
+	pointInTime
+}
+
+type pointInTime struct {
+	floatValue      float64
+	timeValue       time.Time
+	formmattedValue string
+}
+
+type apiQuery struct {
+	domain       string
+	prometheusNS string
+	api          string
+	metricQuery  string
+	port         string
+}
+
+func NewPrometheusTest() (*prometheusTest, error) {
+
+	queryToApi := apiQuery{api: api, domain: domain, metricQuery: metricsQuery, port: port, prometheusNS: prometheusNS}
+
+	return &prometheusTest{
+		metricName: metricName,
+		uuid:       uuid.New().String(),
+		apiQuery:   queryToApi,
+	}, nil
+}
+
+func (point *pointInTime) pointInTime(f float64) {
+	point.floatValue = f
+
+	t := time.Unix(int64(f), 0) //gives unix time stamp in utc
+	point.timeValue = t
+
+	point.formmattedValue = t.Format(time.RFC3339)
+	log.Printf("Point in time: %s", point.formmattedValue)
+}
+
+type Connector interface {
+	connectToPrometheusApi(domain, port, api, query, pointInTime string) error
+}
+
+func (qresp *queryResponse) connectToPrometheusApi(domain, port, api, query, pointInTime string) {
+
+	values := url.Values{}
+	values.Set("query", query)
+	if pointInTime != "" {
+		values.Set("time", pointInTime)
+	}
+
+	uri := domain + ":" + port + api
+	client := &http.Client{
+		Timeout: 15 * time.Second,
+	}
+	url := uri + values.Encode()
+	log.Println(url)
```

Comment:
all lines that you create using log will be shown even when the test does not fail. imho the proper way for such output would be to use the testing object. (which is not yet accessible, so maybe we should somehow pass it to the  end2end tests)